samples/kotlin-android-app-destinations/src/main/java/com/segment/analytics/destinations/plugins/AppsflyerDestination.kt

```diff
@@ -0,0 +1,244 @@
+        appsflyer?.setCurrencyCode(currencyCode)
+        analytics.log("appsflyer.setCurrencyCode($currencyCode)", type = LogType.INFO)
+        appsflyer?.setDebugLog(isDebug)
+        analytics.log("appsflyer.setDebugLog($isDebug)", type = LogType.INFO)
+    }
+
+
+    companion object {
+        const val AF_SEGMENT_SHARED_PREF = "appsflyer-segment-data"
+        const val CONV_KEY = "AF_onConversion_Data"
+    }
+
+    private val propertiesMapper = mapOf(
+        "revenue" to AFInAppEventParameterName.REVENUE,
+        "price" to AFInAppEventParameterName.PRICE,
+        "currency" to AFInAppEventParameterName.CURRENCY
+    )
+
+    interface ExternalAppsFlyerConversionListener : AppsFlyerConversionListener
+    interface ExternalDeepLinkListener : DeepLinkListener
+
+    inner class ConversionListener(val analytics: Analytics) : AppsFlyerConversionListener {
+        override fun onConversionDataSuccess(conversionData: Map<String, Any>) {
+            if (!getFlag(CONV_KEY)) {
+                trackInstallAttributed(conversionData)
+                setFlag(CONV_KEY, true)
+            }
+            conversionListener?.onConversionDataSuccess(conversionData)
+        }
+
+        override fun onConversionDataFail(errorMessage: String) {
+            conversionListener?.onConversionDataFail(errorMessage)
+        }
+
+        override fun onAppOpenAttribution(attributionData: Map<String, String>) {
+            conversionListener?.onAppOpenAttribution(attributionData)
+        }
+
+        override fun onAttributionFailure(errorMessage: String) {
+            conversionListener?.onAttributionFailure(errorMessage)
+        }
+
+        private fun convertToPrimitive(value: Any?): JsonElement {
+            return when (value) {
+                is Boolean -> JsonPrimitive(value)
+                is Number -> JsonPrimitive(value)
+                is String -> JsonPrimitive(value)
+                is Map<*, *> -> buildJsonObject {
+                    value.forEach { (k, v) ->
+                        put(k.toString(), convertToPrimitive(v))
+                    }
+                }
+                is List<*> -> buildJsonArray {
+                    value.forEach { v ->
+                        add(convertToPrimitive(v))
+                    }
+                }
+                is Array<*> -> buildJsonArray {
+                    value.forEach { v ->
+                        add(convertToPrimitive(v))
+                    }
+                }
+                else -> JsonPrimitive(value.toString())
+            }
+        }
+
+        private fun trackInstallAttributed(attributionData: Map<String, Any>) {
+            // See https://segment.com/docs/spec/mobile/#install-attributed.
+            val properties = buildJsonObject {
+                put("provider", key)
+                attributionData.forEach { (k, v) ->
+                    if (k != "media_source" && k != "adGroup") {
```

Comment:
nit: maybe `k !in setOf("media_source", "adGroup")` is better? in case we are excluding other keys in the future.

Reply:
i like that. will change